demos/common/mqtt_demo_helpers/mqtt_demo_helpers.c

```diff
@@ -407,6 +480,146 @@ static TransportSocketStatus_t prvConnectToServerWithBackoffRetries( NetworkCont
+    {
+        /* Event callback will set #usPacketTypeReceived when receiving appropriate packet. This
+         * will wait for at most mqttexamplePROCESS_LOOP_TIMEOUT_MS. */
+        xMQTTStatus = MQTT_ProcessLoop( pxMQTTContext, mqttexamplePROCESS_LOOP_TIMEOUT_MS );
+    }
+
+    if( ( xMQTTStatus != MQTTSuccess ) || ( globalPacketTypeReceived != usPacketType ) )
+    {
+        LogError( ( "MQTT_ProcessLoop failed to receive packet: Packet type=%02X, LoopDuration=%u, Status=%s",
+                    usPacketType,
+                    ( mqttexamplePROCESS_LOOP_TIMEOUT_MS * ucCount ),
+                    MQTT_Status_strerror( xMQTTStatus ) ) );
+    }
+
+    return xMQTTStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+BaseType_t SubscribeToTopic( MQTTContext_t * pxMqttContext,
+                             const char * pcTopicFilter,
+                             uint16_t usTopicFilterLength )
+{
+    MQTTStatus_t xResult = MQTTSuccess;
+    BackoffAlgorithmContext_t xRetryParams;
+    BaseType_t xBackoffStatus = pdFAIL;
+    MQTTSubscribeInfo_t xMQTTSubscription;
+    BaseType_t xFailedSubscribeToTopic = pdFALSE;
+    uint32_t ulTopicCount = 0U;
+    BaseType_t xStatus = pdFAIL;
+
+    assert( pxMqttContext != NULL );
+    assert( pcTopicFilter != NULL );
+    assert( usTopicFilterLength > 0 );
+
+    /* Some fields not used so start with everything at 0. */
+    ( void ) memset( ( void * ) &xMQTTSubscription, 0x00, sizeof( xMQTTSubscription ) );
+
+    /* Initialize the status of the subscription. */
+    xTopicFilterContext.pcTopicFilter = pcTopicFilter;
+    xTopicFilterContext.xSubAckStatus = MQTTSubAckFailure;
+
+    /* Get a unique packet id. */
+    globalSubscribePacketIdentifier = MQTT_GetPacketId( pxMqttContext );
+
+    /* Subscribe to the pcTopicFilter topic filter. This example subscribes to
+     * only one topic and uses QoS1. */
+    xMQTTSubscription.qos = MQTTQoS1;
+    xMQTTSubscription.pTopicFilter = pcTopicFilter;
+    xMQTTSubscription.topicFilterLength = usTopicFilterLength;
+
+    /* Initialize retry attempts and interval. */
+    BackoffAlgorithm_InitializeParams( &xRetryParams,
+                                       SUBSCRIBE_RETRY_BACKOFF_BASE_MS,
+                                       SUBSCRIBE_RETRY_MAX_BACKOFF_DELAY_MS,
+                                       SUBSCRIBE_RETRY_MAX_ATTEMPTS );
+
+    do
+    {
+        /* The client is now connected to the broker. Subscribe to the topic
+         * as specified in pcTopicFilter at the top of this file by sending a
+         * subscribe packet then waiting for a subscribe acknowledgment (SUBACK).
+         * This client will then publish to the same topic it subscribed to, so it
+         * will expect all the messages it sends to the broker to be sent back to it
+         * from the broker. This demo uses QOS0 in Subscribe, therefore, the Publish
+         * messages received from the broker will have QOS0. */
```

Comment:
This comment is not relevant to this file
```suggestion
         */
```

Reply:
My copy pasta instincts have failed me